src/com/google/javascript/jscomp/ProductionCoverageInstrumentationCallback.java

```diff
@@ -91,23 +123,109 @@ public void visit(NodeTraversal traversal, Node node, Node parent) {
+        }
+        Node ifFalseNode = node.getLastChild();
+        if (NodeUtil.isEmptyBlock(ifFalseNode)
+            || (ifFalseNode.getFirstChild() != null && !ifFalseNode.getFirstChild().isIf())) {
+          instrumentBlockNode(ifFalseNode, sourceFileName, functionName, Type.BRANCH_DEFAULT.value);
+        }
+        break;
+      case SWITCH:
+        boolean hasDefaultCase = false;
+        for (Node c = node.getSecondChild(); c != null; c = c.getNext()) {
+          if (c.isDefaultCase()) {
+            instrumentBlockNode(
+                c.getLastChild(), sourceFileName, functionName, Type.BRANCH_DEFAULT.value);
+            hasDefaultCase = true;
+          } else {
+            instrumentBlockNode(c.getLastChild(), sourceFileName, functionName, Type.BRANCH.value);
+          }
+        }
+        if (!hasDefaultCase) {
+          Node defaultBlock = IR.block();
+          defaultBlock.useSourceInfoIfMissingFromForTree(node);
+          Node defaultCase = IR.defaultCase(defaultBlock).useSourceInfoIfMissingFromForTree(node);
+          node.addChildToBack(defaultCase);
+          instrumentBlockNode(
+              defaultBlock, sourceFileName, functionName, Type.BRANCH_DEFAULT.value);
+        }
+        break;
+      case HOOK:
+        Node ifTernaryIsTrueExpression = node.getSecondChild();
+        Node ifTernaryIsFalseExpression = node.getLastChild();
+
+        addInstrumentationNodeWithComma(
+            ifTernaryIsTrueExpression, sourceFileName, functionName, Type.BRANCH.value);
+        addInstrumentationNodeWithComma(
+            ifTernaryIsFalseExpression, sourceFileName, functionName, Type.BRANCH.value);
+
+        compiler.reportChangeToEnclosingScope(node);
+        break;
+      case OR:
+      case AND:
+      case COALESCE:
+        // Only instrument the second child of the binary operation because the first child will
+        // always execute, or the first child is part of a chain of binary operations and would have
+        // already been instrumented.
+        Node secondExpression = node.getLastChild();
+        addInstrumentationNodeWithComma(
+            secondExpression, sourceFileName, functionName, Type.BRANCH.value);
+
+        compiler.reportChangeToEnclosingScope(node);
+        break;
+      default:
+        if (NodeUtil.isLoopStructure(node)) {
+          Node blockNode = NodeUtil.getLoopCodeBlock(node);
+          checkNotNull(blockNode);
+          instrumentBlockNode(blockNode, sourceFileName, functionName, Type.BRANCH.value);
+        }
     }
   }
 
   /**
-   * Iterate over all collected block nodes within a Script node and add a new child to the front of
-   * each block node which is the instrumentation Node
+   * Given a node, this function will create a new instrumentationNode and combine it with the
+   * original node using a COMMA node.
+   */
+  private void addInstrumentationNodeWithComma(
+      Node originalNode, String fileName, String functionName, String type) {
```

Comment:
Pass `Type` as enum, not as a string and then you can get its value only when you need it.